language/vm/src/file_format.rs

```diff
@@ -1642,6 +1657,9 @@ pub struct CompiledModuleMut {
     pub struct_defs: Vec<StructDefinition>,
     /// Function defined in this module.
     pub function_defs: Vec<FunctionDefinition>,
+
+    /// Friend declarations of this module.
+    pub friend_decls: Vec<FriendDeclaration>,
```

Comment:
Nit, I'd move this up with the other tables. It is weird for it to come after them (probably put it right after field handles)

Reply:
Since we are moving this, I think we can put it right after `module_handles` and before  `struct_handles`? The rationale is that the friend list is expected to be declared before any functions and structs are defined.

Reply:
Ignore the comment above. Yes, this list should be after the handles. After `field_handles` is the right place.